spyder/api/widgets/comboboxes.py

```diff
@@ -0,0 +1,250 @@
+# (see spyder/__init__.py for details)
+
+"""
+Spyder combobox widgets.
+
+Use these widgets for any combobox you want to add to Spyder.
+"""
+
+# Standard library imports
+import sys
+
+# Third-party imports
+import qstylizer.style
+from qtpy.QtCore import QSize, Qt, Signal
+from qtpy.QtGui import QColor
+from qtpy.QtWidgets import (
+    QComboBox,
+    QFontComboBox,
+    QFrame,
+    QLineEdit,
+    QStyledItemDelegate
+)
+
+# Local imports
+from spyder.utils.palette import QStylePalette
+from spyder.utils.stylesheet import AppStyle, WIN
+
+
+class _SpyderComboBoxDelegate(QStyledItemDelegate):
+    """
+    Delegate to make separators color follow our theme.
+
+    Adapted from https://stackoverflow.com/a/33464045/438386
+    """
+
+    def paint(self, painter, option, index):
+        data = index.data(Qt.AccessibleDescriptionRole)
+        if data and data == "separator":
+            painter.setPen(QColor(QStylePalette.COLOR_BACKGROUND_6))
+            painter.drawLine(
+                option.rect.left() + AppStyle.MarginSize,
+                option.rect.center().y(),
+                option.rect.right() - AppStyle.MarginSize,
+                option.rect.center().y()
+            )
+        else:
+            super().paint(painter, option, index)
+
+    def sizeHint(self, option, index):
+        data = index.data(Qt.AccessibleDescriptionRole)
+        if data and data == "separator":
+            return QSize(0, 3 * AppStyle.MarginSize)
+
+        return super().sizeHint(option, index)
+
+
+class _SpyderComboBoxLineEdit(QLineEdit):
+    """Dummy lineedit used for non-editable comboboxes."""
+
+    sig_mouse_clicked = Signal()
+
+    def __init__(self, parent):
+        super().__init__(parent)
+
+        # Fix style issues
+        css = qstylizer.style.StyleSheet()
+        css.QLineEdit.setValues(
+            # These are necessary on Windows to prevent some ugly visual
+            # glitches.
+            backgroundColor="transparent",
+            border="none",
+            padding="0px",
+            # Make text look centered for short comboboxes
+            paddingRight=f"-{3 if WIN else 2}px"
+        )
+
+        self.setStyleSheet(css.toString())
+
+    def mouseReleaseEvent(self, event):
+        self.sig_mouse_clicked.emit()
+        super().mouseReleaseEvent(event)
+
+    def mouseDoubleClickEvent(self, event):
+        # Avoid selecting the lineedit text with double clicks
+        pass
+
+
+class _SpyderComboBoxMixin:
+    """Mixin with the basic style and functionality for our comboboxes."""
+
+    def __init__(self):
+
+        # Style
+        self._css = self._generate_stylesheet()
+        self.setStyleSheet(self._css.toString())
+
+    def contextMenuEvent(self, event):
+        # Prevent showing context menu for editable comboboxes because it's
+        # added automatically by Qt. That means that the menu is not built
+        # using our API and it's not localized.
+        pass
```

Comment:
Should then a menu be manually created instead of just passing? WIthout an implementation then the only way you have to copy/paste/cut text is by using shortcuts, no?

Reply:
I talked about this to @conradolandia in one of our UX meetings and we agreed that:

* It's a minor functionality that can also be done using the mouse or keyboard shortcuts (as you said).
* It's not worth it to invest time to implement a new, localized menu that uses our API because we have a ton of other things to do for Spyder 6.